engine/src/main/java/com/datatorrent/stram/client/StramAgent.java

```diff
@@ -417,11 +417,25 @@ private StramWebServicesInfo retrieveWebServicesInfo(String appId)
       String appPath = response.getString("appPath");
       String user = response.getString("user");
       JSONObject permissionsInfo = null;
-      try (FSDataInputStream is = fileSystem.open(new Path(appPath, "permissions.json"))) {
-        permissionsInfo = new JSONObject(IOUtils.toString(is));
+      Path permissionsPath = new Path(appPath, "permissions.json");
+      LOG.debug("Checking for permission information in file {}", permissionsPath);
+      try {
+        if (fileSystem.exists(permissionsPath)) {
+          LOG.info("Loading permission information");
+          try (FSDataInputStream is = fileSystem.open(permissionsPath)) {
+            permissionsInfo = new JSONObject(IOUtils.toString(is));
+          }
+        } else {
+          // ignore and log messages if file is not found
+          LOG.info("Permission information not available, file not present");
+        }
       } catch (IOException ex) {
-        // ignore if file is not found
-        LOG.info("Exception in accessing permissions.json", ex);
+        // ignore and log message when unable to read the file
+        if (LOG.isDebugEnabled()) {
```

Comment:
Looking at the code I don't think that the exception will be logged continuously. In case I am wrong, I still will prefer to log the first exception for any log level and suppress other log entries completely.

Two constructors on PermissionInfo class, one that takes JSON object for cases where JSON is already in memory and one that takes file location for cases where it needs to be loaded from HDFS will allow achieving exactly the functionality you talk about. I don't see why StramAgent is a better place to load the file and convert it to JSON.